src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Unix.cs

```diff
@@ -140,53 +142,56 @@ public partial class NetworkChange
         private static unsafe void CreateSocket()
         {
             Debug.Assert(s_socket == null, "s_socket is not null, must close existing socket before opening another.");
-            int newSocket;
+            IntPtr newSocket;
             Interop.Error result = Interop.Sys.CreateNetworkChangeListenerSocket(&newSocket);
             if (result != Interop.Error.SUCCESS)
             {
                 string message = Interop.Sys.GetLastErrorInfo().GetErrorMessage();
                 throw new NetworkInformationException(message);
             }
 
-            s_socket = new SocketWrapper(newSocket);
+            s_socket = new Socket(new SafeSocketHandle(newSocket, ownsHandle: true));
 
-            new Thread(args => LoopReadSocket((SocketWrapper)args!))
-            {
-                IsBackground = true,
-                Name = ".NET Network Address Change"
-            }.UnsafeStart(s_socket);
+            // Don't capture ExecutionContext.
```

Comment:
> By moving this to thread pool we lost ability to test this.

Can you elaborate?

> I assume we could run out of thread pool threads

Note it's not blocking a thread pool thread.  It's just using one long enough to start the asynchronous operation.

Reply:
The deleted test checked the process for threads named `".NET Network Address Change"` and made sure there are none after the event callback got unregistered. Since the original problem was that we were leaking them. Now, we don't have any coverage while we redo the logic here. So I'm cautious of reintroducing the same problem we had.

You're right, with `await` in the body we're not blocking any thread. But I guess that instead of threads we'd leak async state machines or rather `SockerAsyncEventArgs` or whatever is used underneath.